src/provider.tsx

```diff
@@ -226,6 +214,8 @@ export const UseInkathonProvider: FC<UseInkathonProviderProps> = ({
     unsubscribeAccounts?.()
     setUnsubscribeAccounts(undefined)
     setActiveExtension(undefined)
+    const adapter = await getNightlyConnectAdapter()
+    await adapter?.disconnect()
```

Comment:
I would suggest we wrap this in an if-block, whether `activeExtension.name === nightlyConnect.id` (untested).

Reply:
unfortunately activeExtension is undefined in the disconnect function